examples/simple.rs

```diff
@@ -29,16 +33,52 @@ async fn main() -> Result<(), Report> {
     //
     // Falling back to the regular tracing library.
     tracing_subscriber::fmt::fmt()
-        .with_max_level(tracing::Level::DEBUG)
+        .with_env_filter("trauma=debug")
         .init();
 
     // Download the file(s).
     let reqwest_rs = "https://github.com/seanmonstar/reqwest/archive/refs/tags/v0.11.9.zip";
-    let downloads = vec![Download::try_from(reqwest_rs).unwrap()];
+    let fake = format!("{}.fake", &reqwest_rs);
+    let downloads = vec![
+        Download::try_from(reqwest_rs).unwrap(),
+        Download::try_from(fake.as_str()).unwrap(),
+    ];
     let downloader = DownloaderBuilder::new()
         .directory(PathBuf::from("output"))
         .build();
-    downloader.download(&downloads).await;
+    let summaries = downloader.download(&downloads).await;
+
+    // Display results.
+    display_summary(&summaries);
 
     Ok(())
 }
+
+fn display_summary(summaries: &[Summary]) {
+    let mut table = Table::new();
+    let header = Row::from(vec!["File", "Size", "Status", "Error"]);
+    table.set_header(header);
+    summaries.iter().for_each(|s| {
+        let mut error = String::new();
```

Comment:
It's good to avoid allocating a string on the heap unless we are guaranteed to use it, in this case `error` will only be pushed to if `s.status()` matches `Status::Fail(_)`, better to define it as `Option<String>`:
```rust
fn display_summary(summaries: &[Summary]) {
    let mut table = Table::new();
    let header = Row::from(vec!["File", "Size", "Status", "Error"]);
    table.set_header(header);
    summaries.iter().for_each(|s| {
        let mut error: Option<String> = None;
        let status = match s.status() {
            Status::Success => '✅',
            Status::Fail(msg) => {
                if msg.len() >= 50 {
                    let mut msg = msg.to_owned();
                    msg.truncate(50);
                    msg.push('…'); // using the unicode char '…'
                    error = msg.into();
                } else {
                    error = msg.to_owned().into();
                }
                '❌'
            }
            Status::NotStarted => '🔜',
        };
        table.add_row(vec![
            &s.download().filename,
            &s.size().to_string(),
            &status.to_string(),
            &error.unwrap_or_default(),
        ]);
        ()
    });
    println!("{table}");
}
```

Reply:
One benefit of having `error` as `Option<_>` is that we can conditionally create the `"Error"` row if need be.
https://github.com/rgreinho/trauma/blob/5df8b08f092f044989d1d9685b7d77224a4fd6c2/examples/simple.rs#L59
```rust
fn display_summary(summaries: &[Summary]) {
    let mut table = Table::new();
    let mut header = Row::from(vec!["File", "Size", "Status"]);
    summaries.iter().for_each(|s| {
        let mut error: Option<String> = None;
        // ...
        let mut row = vec![
            s.download().filename.clone(),
            s.size().to_string(),
            status.to_string(),
        ];
        error.map(|msg| {
            if header.cell_count() == 3 {
                header.add_cell("Error".into());
            }
            row.push(msg);
        });
        table.add_row(row);
        ()
    });
    table.set_header(header);
    println!("{table}");
}
```

Reply:
:P